projects/packages/forms/src/dashboard/state/inbox/reducer.js

```diff
@@ -0,0 +1,70 @@
+/**
+ * External dependencies
+ */
+import { map } from 'lodash';
+import { combineReducers } from 'redux';
+/**
+ * Internal dependencies
+ */
+import {
+	JETPACK_FORMS_RESPONSES_FETCH,
+	JETPACK_FORMS_RESPONSES_FETCH_RECEIVE,
+	JETPACK_FORMS_RESPONSES_FETCH_FAIL,
+} from '../action-types';
+
+export const loading = ( state = false, action ) => {
+	if ( action.type === JETPACK_FORMS_RESPONSES_FETCH ) {
+		return true;
+	}
+
+	if (
+		action.type === JETPACK_FORMS_RESPONSES_FETCH_RECEIVE ||
+		action.type === JETPACK_FORMS_RESPONSES_FETCH_FAIL
+	) {
+		return false;
+	}
+
+	return state;
+};
+
+export const responses = ( state = [], action ) => {
+	if ( action.type === JETPACK_FORMS_RESPONSES_FETCH && action.offset === 0 ) {
+		return [];
+	}
+
+	if ( action.type === JETPACK_FORMS_RESPONSES_FETCH_RECEIVE ) {
+		return [ ...( action.append ? state : [] ), ...action.responses ];
```

Comment:
Hmm, this seems like an appropriate implementation for infinite scroll. Are we pursuing that now instead of pagination?

For pagination, it'd actually cause quite a few problems and I'd argue at least initially it's best to only store the results for the current page/query. No appending.

Reply:
Yeah, I see that now. I merely copied all the state/store things from your old PR, I think at that time we were still considering infinite scroll. I'll change it.

Reply:
I see, thanks 🙇‍♂️